sdk/typescript/src/builder/__tests__/Transaction.test.ts

```diff
@@ -11,6 +11,23 @@ it('can construct and serialize an empty tranaction', () => {
 	expect(() => tx.serialize()).not.toThrow();
 });
 
+it('can construct a receiving transaction argument', () => {
+	const tx = new TransactionBlock();
+	tx.receiving(Inputs.ReceivingRef(ref()));
+	expect(() => tx.serialize()).not.toThrow();
+});
+
+it('receiving transaction argument different from object argument', () => {
+	const oref = ref();
+	const rtx = new TransactionBlock();
+	rtx.receiving(Inputs.ReceivingRef(oref));
+	const otx = new TransactionBlock();
+	otx.receiving(Inputs.ObjectRef(oref));
```

Comment:
This test actually highlights the thing that I was afraid of (that it's legal to pass an `ObjectRef` into a `receiving` and vice versa).  Why do we have both a `receiving` function and a `ReceivingRef` input?

Reply:
`receiving` is there purely as a convenience, but is not necessary -- what really determines the argument type is either:
* The ref type passed in (if you pass a reference type), or
* The resolution based on type arguments/object types returned from underlying RPC calls (in the case where you only pass an object ID). 

Happy to remove `receiving` and will do so now :)

Reply:
Ah got it, thank you!  +1 happy to not give people the choice here :)